src/main/webapp/app/entities/search/search.util.ts

```diff
@@ -0,0 +1,221 @@
+import { SearchResult } from './search.model';
+
+const createDimensions = (features: readonly IFeature[], dimensionsList: Array<string>) => {
+  const dimensions = [];
+  dimensionsList.forEach(element => {
+    const dimension = features.find(item => {
+      return item.name === trim(element);
+    });
+    if (dimension) {
+      dimensions.push(dimension);
+    }
+  });
+  return dimensions;
+};
+
+const createMeasures = (features: readonly IFeature[], measuresList: Array<string>) => {
+  const measures = [];
+  measuresList.forEach(element => {
+    const measure = features.find(item => {
+      const featureName = element.substring(element.indexOf('(') + 1, element.indexOf(')'));
+      return item.name === trim(featureName);
+    });
+    if (measure) {
+      measures.push(measure);
+    }
+  });
+  return measures;
+};
+
+const setDimensions = (dimensions: readonly IFeature[], visual: IVisualMetadataSet, visualWrap: any) => {
+  dimensions.forEach((element, index) => {
+    if (index === 0) {
+      visual.fields[index].feature = dimensions[index];
+    } else {
+      const fieldType = visualWrap.nextFieldDimension(visual.fields, visual.metadataVisual);
+      const field = {
+        fieldType,
+        feature: dimensions[index],
+        constraint: fieldType.constraint,
+        properties: fieldType.propertyTypes.map(function (item) {
+          return {
+            propertyType: item.propertyType,
+            value: item.propertyType.defaultValue,
+            type: item.propertyType.type,
+            order: item.order,
+          };
+        }),
+        order: fieldType.order,
+      };
+      visual.fields.push(field);
+    }
+  });
+  return visual;
+};
+
+const setMeasures = (measures: readonly IFeature[], visual: IVisualMetadataSet, visualWrap: any) => {
+  measures.forEach((element, index) => {
+    const fieldType = visualWrap.nextFieldMeasure(visual.fields, visual.metadataVisual);
+    const field = {
+      fieldType,
+      feature: measures[index],
+      constraint: fieldType.constraint,
+      properties: fieldType.propertyTypes.map(function (item) {
+        return {
+          propertyType: item.propertyType,
+          value: item.propertyType.defaultValue,
+          type: item.propertyType.type,
+          order: item.order,
+        };
+      }),
+      order: fieldType.order,
+    };
+    visual.fields.push(field);
+  });
+  return visual;
+};
+
+export const createVisualFields = (
+  features: readonly IFeature[],
+  dimensionsList: Array<string>,
+  measuresList: Array<string>,
+  view: IViews,
+  visualization: IVisualizations
+) => {
+  let visual = createVisualMetadata(visualization, view, 0);
+  const dimensions = createDimensions(features, dimensionsList);
+  const measures = createMeasures(features, measuresList);
+  const visualWrap = VisualWrap(visual);
+  visual = setDimensions(dimensions, visual, visualWrap);
+  visual = setMeasures(measures, visual, visualWrap);
+  return visual;
+};
```

Comment:
this seems to be used in multiple places?...can you use the one you created for visualisation

Reply:
yes I have already use visualisation that one is createVisualMetadata 
this one is use for create QueryDTO from string

Reply:
OK..it this different from visualisation

Reply:
yes